src/helm/proxy/clients/openai_client.py

```diff
@@ -187,8 +192,91 @@ def do_it():
             embedding=embedding,
         )
 
+    @staticmethod
+    def _get_tokenizer_name(tokenizer: str) -> str:
+        return "/".join(tokenizer.split("/")[1:])
+
     def tokenize(self, request: TokenizationRequest) -> TokenizationRequestResult:
-        raise NotImplementedError("Use the HuggingFaceClient to tokenize.")
+        # For reproducibility purposes, we use huggingface/gpt2 as the default tokenizer.
+        # except for gpt-3.5 turbo models and gpt-4 that use the tiktoken library.
+
+        if "openai/" not in request.tokenizer:
+            result = self.tokenizer_client.tokenize(
+                # We're assuming the model uses the GPT-2 tokenizer.
+                TokenizationRequest(request.text, tokenizer="huggingface/gpt2")
+            )
+            return result
+        elif request.tokenizer != "openai/cl100k_base":
+            raise ValueError(
+                f"{request.tokenizer} is not supported."
+                + "Only openai/cl100k-base is supported by the tiktoken library for now."
+            )
+
+        cache_key = asdict(request)
+
+        try:
+
+            def do_it():
+                tokenizer = tiktoken.get_encoding(self._get_tokenizer_name(request.tokenizer))
+                # NOTE(josselin): The tokenizer.encode method returns a list of integers.
+                # This shouuld be fine as TokenizationToken accepts both integers and strings.
+                # In reality, some metrics assume strings, so we decode the integers here.
+                # This is Hacky and should be fixed.
+                # If fixed, the following line should be fine:
+                # tokenizer.encode(request.text)
+                tokens = [tokenizer.decode([token]) for token in tokenizer.encode(request.text)]
+                return {"tokens": tokens}
+
+            response, cached = self.cache.get(cache_key, wrap_request_time(do_it))
+
+            result = TokenizationRequestResult(
+                success=True,
+                cached=cached,
+                text=request.text,
+                tokens=[TokenizationToken(value) for value in response["tokens"]],
+                request_time=response["request_time"],
+                error=None,
+            )
+            return result
+        except Exception as error:
+            raise ValueError(
+                f"Error encoding with tiktoken: {error}." + "You might want to consider using huggingface/gpt2."
```

Comment:
I'm curious if this caught any interesting tiktoken errors while you were testing?

Reply:
No, I never had any issues (except typos for encoder names. But better safe than sorry no?

Reply:
Sounds good.